src/Adapter/AmqpLib/Exchange/AmqpExchange.php

```diff
@@ -0,0 +1,142 @@
+
+    /**
+     * @var ExchangeDefinition
+     */
+    private $definition;
+
+    /**
+     * @param AmqpChannel        $channel
+     * @param ExchangeDefinition $definition
+     */
+    public function __construct(AmqpChannel $channel, ExchangeDefinition $definition)
+    {
+        $this->channel = $channel;
+        $this->definition = $definition;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getChannel(): ChannelInterface
+    {
+        return $this->channel;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName(): string
+    {
+        return $this->definition->getName();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function publish(MessageInterface $message, string $routingKey = ''): void
+    {
+        $headers = $message->getHeaders()->all();
+        $identifier = $message->getIdentifier();
+
+        $options = [
+            'content_type'  => $message->getPayload()->getContentType(),
+            'delivery_mode' => $message->getOptions()->isPersistent() ? 2 : 1,
+        ];
+
+        if ($message->getOptions()->getExpiration()) {
+            $options['expiration'] = $message->getOptions()->getExpiration();
+        }
+
+        if ($message->getPayload()->getContentEncoding()) {
+            $options['content_encoding'] = $message->getPayload()->getContentEncoding();
+        }
+
+        if ($identifier->getId()) {
+            $options['message_id'] = $identifier->getId();
+        }
+
+        if ($identifier->getUserId()) {
+            $options['user_id'] = $identifier->getUserId();
+        }
+
+        if ($identifier->getAppId()) {
+            $options['app_id'] = $identifier->getAppId();
+        }
+
+        if (\count($headers)) {
+            $options['application_headers'] = new AMQPTable($headers);
+        }
+
+        $amqplibMessage = new AMQPMessage($message->getPayload()->getData(), $options);
+        $this->channel->getChannel()->basic_publish($amqplibMessage, $this->getName(), $routingKey);
+    }
+
+    /**
+     * Declares exchange (creates or validates existing one)
+     */
+    public function declare()
+    {
+        // Default exchange always exists and "declare" call is not permitted
+        if (!$this->getName()) {
+            return;
+        }
+
+        $arguments = new AMQPTable();
+
+        foreach ($this->definition->getArguments() as $argument) {
+            $arguments->set($argument->getName(), $argument->getValue());
+        }
+
+        $this->channel->getChannel()->exchange_declare(
+            $this->getName(),
+            $this->definition->getType(),
+            $this->definition->isPassive(),
+            $this->definition->isDurable(),
+            false,
+            false,
+            false,
+            $arguments
+        );
+    }
+
+    /**
+     * @param string $exchangeName
+     * @param string $routingKey
+     */
+    public function bind(string $exchangeName, string $routingKey)
```

Comment:
1. `void` return type.
2. This method does not declared in interface. Why method must be here with `public`?